compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1439,7 +1439,11 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         );
 
         if macro_kind == MacroKind::Bang && ident.name == sym::macro_rules {
-            err.subdiagnostic(MaybeMissingMacroRulesName { span: ident.span });
+            let primary_span = ident.span;
+            let label_span = primary_span.with_hi(primary_span.hi() + BytePos(1)).shrink_to_hi();
```

Comment:
Manual manipulation of spans can cause issues in situations where we recover in the lexer. I don't think this diagnostic would look a lot worse if we just used `primary_span.shrink_to_hi()`.

Alternatively, find a more "pure" way to express this span (or a similar one) without using bytepos.

Reply:
>  if we just used primary_span.shrink_to_hi()

This is also looks good to me.  And it is more consistent between `macro_rules! {}` and `macro_rules!{}`, which will always point to the `!`

>  Alternatively, find a more "pure" way to express this span (or a similar one) without using bytepos.
I currently have few time to try this.